server/hotreloadmiddleware/process-update.js

```diff
@@ -33,14 +33,8 @@ module.exports = function(hash, moduleMap, options) {
       if (err) return handleError(err);
 
       if(!updatedModules) {
-        if (options.warn) {
```

Comment:
This is the offending block that was causing issues for us. Particularly line 40. Hotreload was built (in general) to operate on a single bundle, on a single webpage. 

This block says "If there are no updated modules, go ahead and reload anyway, something must have gone wrong."

This is great in an SPA. In finsemble, where we have many components all connected to the same hotReload server, this causes bad behavior. 

Example:
I change a file in the toolbar. Toolbar reloads as it should.
App launcher also gets the event that a file was rebuilt. But it gets into this block. None of my app launcher files changed...but there was a change event. Something must be wrong. reload.

Reload, reload, reload, addEventListener, addEventListener, addEventListener - memory leak, crash.